extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesTaskRunner.java

```diff
@@ -0,0 +1,571 @@
+                        StandardCharsets.UTF_8
+                    );
+                    taskLogPusher.pushTaskLog(task.getId(), log.toFile());
+                  }
+                  finally {
+                    Files.deleteIfExists(log);
+                  }
+                  client.cleanUpJob(new K8sTaskId(task.getId()));
+                  synchronized (tasks) {
+                    tasks.remove(task.getId());
+                  }
+                }
+              })
+          ));
+      return tasks.get(task.getId()).getResult();
+    }
+  }
+
+  JobResponse monitorJob(Task task, K8sTaskId k8sTaskId)
+  {
+    return monitorJob(client.getMainJobPod(k8sTaskId), task, k8sTaskId);
+  }
+
+  JobResponse monitorJob(Pod peonPod, Task task, K8sTaskId k8sTaskId)
+  {
+    if (peonPod == null) {
+      throw new ISE("Error in k8s launching peon pod for task %s", k8sTaskId);
+    }
+    log.info("monitorJob: Enable TLS Port is " + node.isEnableTlsPort());
+    log.info("Pod Host IP " + peonPod.getStatus().getPodIP());
+    TaskLocation location = TaskLocation.create(
+        peonPod.getStatus().getPodIP(),
+        DruidK8sConstants.PORT,
+        DruidK8sConstants.TLS_PORT,
+        node.isEnableTlsPort()
+    );
+    client.waitForProcessToStart(location, 10, TimeUnit.MINUTES);
+
+    updateLocation(task, location);
+    updateStatus(task, TaskStatus.running(task.getId()));
+
+    JobResponse response = client.waitForJobCompletion(
+        k8sTaskId,
+        KubernetesTaskRunnerConfig.toMilliseconds(k8sConfig.maxTaskDuration),
+        TimeUnit.MILLISECONDS
+    );
+    response.setLocation(location);
+    return response;
+  }
+
+  @VisibleForTesting
+  void updateStatus(Task task, TaskStatus status)
+  {
+    TaskRunnerUtils.notifyStatusChanged(listeners, task.getId(), status);
+  }
+
+  @VisibleForTesting
+  void updateLocation(Task task, TaskLocation location)
+  {
+    TaskRunnerUtils.notifyLocationChanged(listeners, task.getId(), location);
+  }
+
+  @Override
+  public void shutdown(String taskid, String reason)
+  {
+    client.cleanUpJob(new K8sTaskId(taskid));
+  }
+
+
+  @Override
+  public Optional<InputStream> streamTaskReports(String taskid)
+  {
+    return Optional.absent();
+  }
+
+  @Override
+  public List<Pair<Task, ListenableFuture<TaskStatus>>> restore()
+  {
+    return ImmutableList.of();
+  }
+
+  @Override
+  public void start()
+  {
+    log.info("Starting cleanup executor for jobs older than 1 day....");
+    cleanupExecutor.scheduleAtFixedRate(
+        () ->
+            client.cleanCompletedJobsOlderThan(
+                KubernetesTaskRunnerConfig.toMilliseconds(k8sConfig.taskCleanupDelay),
+                TimeUnit.MILLISECONDS
+            ),
+        1,
+        KubernetesTaskRunnerConfig.toMilliseconds(k8sConfig.taskCleanupInterval),
+        TimeUnit.MILLISECONDS
+    );
+    log.info("Started cleanup executor for jobs older than 1 day....");
```

Comment:
Debug? Here and above? Probably not very useful for the typical Druid user...

Reply:
sure thing.